testsuite/objects/__init__.py

```diff
@@ -63,6 +65,41 @@ def to_dict(self):
         return {"ttl": self.ttl, "key": self.key.to_dict()}
 
 
+class JsonProperty(Value):
+    """
+    https://github.com/kuadrant/authorino/blob/v0.13.0/api/v1beta1/auth_config_types.go#L89
+    For use in identity extended_property functionality
+    """
+
+    __slots__ = ("name",)
+
+    # pylint: disable=invalid-name
+    def __init__(self, name, value=None, jsonPath=None):
+        super().__init__(value, jsonPath)
+        self.name = name
+
+    def to_dict(self):
+        """Returns dict representation of itself (shallow copy only)"""
+        return {"name": self.name, **super().to_dict()}
+
+
+@dataclass
+class ExtendedProperty(JsonProperty):
+    """
+    Class defining ExtendedProperty which has more features than regular JsonProperty
+    """
+
+    __slots__ = ("overwrite",)
```

Comment:
Why? You can extend dataclasses and specify additional parameters as usual

Reply:
Addressed